swarm/network/kademlia.go

```diff
@@ -89,6 +99,7 @@ type Kademlia struct {
 	nDepth     int      // stores the last neighbourhood depth
 	nDepthC    chan int // returned by DepthC function to signal neighbourhood depth change
 	addrCountC chan int // returned by AddrCountC function to signal peer count change
+	Pof        func(pot.Val, pot.Val, int) (int, bool)
```

Comment:
What is the point of exposing the Pof field in Kademlia struct, and it is set to a global variable pof in NewKademlia function. Is Kademlia.Pof needed outside of this package? If it is needed, why exposing it as a struct field if it is always the same as package global variable? I do not see that Kademlia.Pof is used anywhere in this PR or in elsewhere in the current codebase.

Reply:
It's used for the PSS neighbourhood calculations, which is a different PR.

Reply:
Shouldn't be this change only be relevant to a different PR only? I think that code hygiene is nice to have when merging to master. What if this implementation changes in a different PR?